src/lib/blkalloc/varsize_blk_allocator.cpp

```diff
@@ -614,10 +616,14 @@ blk_count_t VarsizeBlkAllocator::alloc_blks_direct(blk_count_t nblks, blk_alloc_
                 cur_blk_id = b.start_bit + b.nbits;
             }
         }
-        if (++portion_num == get_num_portions()) { portion_num = 0; }
-        BLKALLOC_LOG(TRACE, "alloc direct unable to find in prev portion, searching in portion={}, start_portion={}",
-                     portion_num, m_start_portion_num);
-    } while (nblks_remain && (portion_num != m_start_portion_num) && !hints.is_contiguous && out_blkid.has_room());
```

Comment:
I think the reason is_contigous is here is we forget to remove it after min_blks is being added that already takes care of is_contigous, e.g. either alloca all of nblks or nothing from current portion.